src/dom/DomEvent.js

```diff
@@ -172,15 +172,19 @@ L.DomEvent = {
 	},
 
 	getMousePosition: function (e, container) {
+		var evt = e;
+		if (evt.touches) {
+			evt = evt.touches[0] || evt.changedTouches[0];

```

Comment:
Or maybe make sub-cases of `_fireDOMEvent` named `_fireMouseEvent`, `_fireTouchEvent` and so on.